docs/feature-store/sources-targets.md

```diff
@@ -0,0 +1,47 @@
+(sources-targets)=
+# Sources and targets
```

Comment:
Since sources and targets are so central to the feature-store, I would:
1. remove the Sources and Targets H1 heading, and instead gives sources and targets a H1 heading
2. Move  "Partitioning on Parquet target" into "Parquet target" section, where partitioning will be discussed (maybe we will add more special info on other targets as the time goes)